### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by ycombinator** [elastic/elastic-package#75]
Now that the `elastic-package` tool is gaining maturity in some capabilities/commands and is adding new capabilities/commands quite quickly, we should start adding appropriate documentation for the tool. In particular, I think we need documentation that covers these areas:
- An introduction to the tool, covering its use cases, capabilities, and commands.
- Reference documentation for each command. Ideally this will be available as part of the help text for each command, i.e. when `elastic-package help <command>` is run but also be available in the repo itself.
- Use case or HOWTO guides, e.g. how do I test my package?

**Comment by mtojek** [comment]
@ycombinator I think you can mark some action items as done.

**Comment by ycombinator** [comment]
Thanks @mtojek, I marked the first one as done. I think that's the only one for now.

**Comment by ycombinator** [comment]
@mtojek I'd like to email the packages list talking about the testing features in `elastic-package`. Can you write the HOWTO for pipeline testing? Then we can include both HOWTOs (pipeline and system testing) in the email. Thanks!

**Comment by mtojek** [comment]
It looks like there is one item remaining:

>  Reference documentation for each command. Ideally this will be available as part of the help text for each command, i.e. when elastic-package help <command> is run but also be available in the repo itself.

Currently the documentation for commands is limited to a couple of sentences (see: https://github.com/elastic/elastic-package#commands) without going deeper into explanation for switches and arguments. I would be for leaving it as is and
 adding a paragraph to  `elastic-package help <command>` (Markdown docs can get outdated really quickly). For instance, the `export` command may need more words to describe its actions:

```
elastic-package help export
Use export command to export assets relevant for the package from the Elastic stack.

Usage:
  elastic-package export [command]

Available Commands:
  dashboards  Export dashboards from Kibana

Flags:
  -h, --help   help for export

Global Flags:
  -v, --verbose   verbose mode

Use "elastic-package export [command] --help" for more information about a command.
``` 

@ycombinator would you mind sharing your idea for this bullet point?

**Comment by ycombinator** [comment]
@mtojek Yes, we are mostly in agreement. My thinking was that we should flesh out the help text in the command help itself, which is exactly what you are thinking too:

>  I would be for leaving it as is and adding a paragraph to `elastic-package help <command>`

On top of that, I was thinking we could have a script that is run via `make update` (or similar target) which calls `elastic-package help <command>` for each command and then collects the output in some sort of Markdown documentation file. This file should be committed into the repo and linked from the README (or could even just be part of the README even if that results in a better developer experience + isn't too difficult to implement). This way users can get the same help text both while looking at the repo or while interacting with the tool and the automation ensures that both stay in sync at all times (`make check` should fail if they are not in sync).

**Comment by mtojek** [comment]
I think we're on the same page. Ultimately we'll end up with parsing README.md to find the `Commands` section (https://github.com/elastic/elastic-package#commands) and regenerate it using `elastic-package help <command>`s.

I'm not sure if this should have a high priority in the next release cycle as it looks more like an improvement of user experience to me.

**Comment by ycombinator** [comment]
> I'm not sure if this should have a high priority in the next release cycle as it looks more like an improvement of user experience to me.

Yeah, I tend to agree. So far I think the feedback we're getting about `elastic-package` has been more about the _functionality_ and less about _how_ to use the tool. I'd be in favor of scheduling this issue for the next release cycle but as a lower priority. We could also split it up into two smaller tasks so we can make _some_ progress even if we can't completely resolve this issue in one cycle: one task to flesh out the help text for each command and another for the script/automation to regenerate the online documentation.

**Comment by mtojek** [comment]
I've some spare time, I can start filling up descriptions.

**Comment by ycombinator** [comment]
There is one item left to complete in the checklist for this issue:

> Reference documentation for each command. Ideally this will be available as part of the help text for each command, i.e. when `elastic-package help <command>` is run but also be available in the repo itself.

I'm going to take a stab at this now.

**Comment by ycombinator** [comment]
All checklist items for this issue have now been completed. Closing.
